### PR TITLE
Python 3 support (spyne >=2.12), added tests, some cleanup

### PR DESCRIPTION
**Pull request by mj026**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -37,10 +40,11 @@
     install_requires=[
         'setuptools',
         'spyne>=2.12',
-        'lxml'
+        'lxml',
```
Comment (specialunderwear) [inline setup.py]: why are we depending on the dreaded lxm?

File: setup.py [hunk 2]
```diff
@@ -37,10 +40,11 @@
...
+        'six'
```
Comment (specialunderwear) [inline setup.py]: do you like six?

File: spynedelegate/meta.py [hunk 3]
```diff
@@ -104,6 +101,7 @@ def __new__(cls, name, bases, attrs):
         return type.__new__(cls, name, bases, delegate_attrs)
 
 
+@six.add_metaclass(DelegateMetaClass)
```
Comment (specialunderwear) [inline spynedelegate/meta.py:104]: Can we get any votes on python3? @all @universe @cryforhelp

File: tests/base.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    @classmethod
+    def setUpClass(cls):
+        queue = multiprocessing.Queue()
+        cls.server = multiprocessing.Process(
+            target=cls.spyne_server, args=(queue,))
+        cls.server.start()
+
+        # wait for wsgi server to start
+        queue.get('server started')
+
```
Comment (specialunderwear) [inline tests/base.py:44]: NICE TRICKS HERE!!!

File: spynedelegate/meta.py [hunk 5]
```diff
@@ -159,8 +155,9 @@ def __init__(cls, name, bases, attrs):  # noqa
                 DelegateServiceMetaClass, cls).__init__(name, bases, attrs)
 
 
+@six.add_metaclass(DelegateServiceMetaClass)
```
Comment (specialunderwear) [inline spynedelegate/meta.py:158]: I don't like python 3

**Review by specialunderwear — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (mj026) [reply setup.py]: Because the Soap11 service needs it. I moved it to the tests dependencies where it should be

File: spynedelegate/meta.py (hunk 3 above)
Comment (mj026) [reply spynedelegate/meta.py:104]: I really would like this to have python 3 support.

File: spynedelegate/meta.py (hunk 5 above)
Comment (mj026) [reply spynedelegate/meta.py:158]: But I like it.

**Pull request merged**